// Copyright 2020 The gRPC Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package grpc.autogpt;

message AIInfo {
  string name = 1;
  string role = 2;
  repeated string goals = 3;
}

message AIRes {
  string thoughts = 1;
  string reasoning = 2;
  string plan = 3;
  string criticism = 4;
  string next_action = 5;
  string system_res = 6;
}

message AutogptRequest {
  AIInfo  ai_info = 1;
}

message AutogptResponse {
  oneof autogpt_response {
    AIRes ai_res = 1;
  }
}

service  Autogpt{
  // Makes a phone call and communicate states via a stream.
  rpc autogptService(stream AutogptRequest) returns (stream AutogptResponse);
}
